### PR TITLE
The osbuild and otk documentation expansion initiative

### PR DESCRIPTION
**Pull request by supakeen**
Based on #77. Initial version is done; can be expanded later.

### CODE REVIEW

File: docs/developer-guide/02-projects/index.md [hunk 1]
```diff
@@ -1 +1,51 @@
...
+
+---
+
+## OSBuild
+
+[OSBuild](./osbuild) is the lowest level in our stack of tools. It takes JSON manifests that describe artifacts and builds those artifacts. `OSBuild` aims to make the building of operating system artifacts precise, declarative, and reproducible. `OSBuild` is written in Python.
+
+### Omnifest Toolkit
+
+[otk](./otk) is a tool to transform files called omnifests to inputs for image build tooling; for example [osbuild](./osbuild) manifests. `otk` aims to make it easier to write definitions for distribution maintainers. `otk` is written in Python.
```
Comment (mvo5) [inline docs/developer-guide/02-projects/index.md:13]: Given that this is still WIP I would rephrase it (and maybe move last?). Something like: `A new otk tool is being worked on to transform a high level image definition into an osbuild manifest, more details about the status of the discussion/implementation can be found in <link-to-otk-repo>` or something along those lines.

**Review by mvo5 — COMMENTED** [review]
I like that this adds a lot of useful information. However I would leave otk documentation for now in the `otk` repo until both the spec and the implementation are more complete.

**Comment by supakeen** [comment]
I'll be removing the `otk` parts from this PR as they are now re-imported: https://github.com/osbuild/otk/pull/49

We'll probably move these back once `otk` is more than a poc.

**Comment by supakeen** [comment]
Going to resubmit this with only the osbuild parts.
